### PR TITLE
Provisioning user signup

### PR DESCRIPTION
**Pull request by ranakan19**
## Description
adding a condition type and condition reason to indicate usersignup provisioning is in progress.

## Checks
1. Did you run `make generate` target? **yes**

2. Did `make generate` change anything in other projects (host-operator, member-operator)? **no**

3. In case of **new** CRD, did you the following? **NA**
    - make/generate.mk in this repository
    - `resources/setup/roles/host.yaml` in the sandbox-sre repository
    - `PROJECT` file: https://github.com/codeready-toolchain/host-operator/blob/master/PROJECT
    - `CSV` file: https://github.com/codeready-toolchain/host-operator/blob/master/config/manifests/bases/host-operator.clusterserviceversion.yaml

4. In case other projects are changed, please provides PR links. - **NA**

### CODE REVIEW

**Review by alexeykazakov — APPROVED** [review]

File: api/v1alpha1/usersignup_types.go [hunk 1]
```diff
@@ -11,6 +11,8 @@ const (
 	UserSignupApproved ConditionType = "Approved"
 	// UserSignupComplete means provisioning is complete
 	UserSignupComplete ConditionType = "Complete"
+	//UserSignupProvisioning means provisioning is in progress
+	UserSignupProvisioning ConditionType = "Provisioning"
```
Comment (MatousJobanek) [inline api/v1alpha1/usersignup_types.go:15]: we don't need a new condition type for this particular reason - it should have been set in the `Complete` one

**Review by MatousJobanek — CHANGES_REQUESTED** [review]
Please drop the new ConditionType - we don't need that

File: api/v1alpha1/usersignup_types.go (hunk 1 above)
Comment (ranakan19) [reply api/v1alpha1/usersignup_types.go:15]: yeah that makes sense! I'll just keep the reason

Comment (alexeykazakov) [reply api/v1alpha1/usersignup_types.go:15]: Ah, right! I overlooked that. I was thinking you Kanika are introducing a new reason. Not condition!

Comment (ranakan19) [reply api/v1alpha1/usersignup_types.go]: done in ec82f43

**Review by MatousJobanek — APPROVED** [review]
Thanks

**Pull request merged**